src/Sylius/Bundle/ShippingBundle/Form/EventListener/AddCodeFormSubscriber.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ShippingBundle\Form\EventListener;
+
+use Sylius\Component\Resource\Model\CodeAwareInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Intl\Exception\UnexpectedTypeException;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class AddCodeFormSubscriber implements EventSubscriberInterface
```

Comment:
This class should be in the ResourceBundle. :) It is not coupled to Shipping at all.